crates/server/src/authority/auth_lookup.rs

```diff
@@ -106,6 +109,26 @@ impl LookupObject for AuthLookup {
         let additionals = Self::take_additionals(self);
         additionals.map(|a| Box::new(a) as Box<dyn LookupObject>)
     }
+
+    #[cfg(feature = "dnssec")]
+    fn dnssec_summary(&self) -> DnssecSummary {
+        let mut all_secure = None;
+        for record in self {
+            match record.proof() {
+                Proof::Secure => {
+                    all_secure.get_or_insert(true);
+                }
+                Proof::Bogus => return DnssecSummary::Bogus,
+                _ => all_secure = Some(false),
+            }
+        }
+
+        if all_secure.unwrap_or(false) {
+            DnssecSummary::Secure
+        } else {
+            DnssecSummary::Insecure
+        }
```

Comment:
I'd write this as:

```rust
match all_secure {
    Some(true) => DnssecSummary::Secure,
    _ => DnssecSummary::Insecure,
}
```

This is a little more compact and has less indirection.